zanata-war/src/main/java/org/zanata/action/MergeTransAction.java

```diff
@@ -0,0 +1,230 @@
+         */
+        if(sourceProjectSlug == null) {
+            setSourceProjectSlug(targetProjectSlug);
+        }
+    }
+
+    public void setSourceProjectSlug(String sourceProjectSlug) {
+        if(!StringUtils.equals(this.sourceProjectSlug, sourceProjectSlug)) {
+            this.sourceProjectSlug = sourceProjectSlug;
+            sourceProject = null;
+            if (!getSourceProject().getProjectIterations().isEmpty()) {
+                sourceVersionSlug =
+                        getSourceProject().getProjectIterations().get(0)
+                                .getSlug();
+            }
+        }
+    }
+
+    public HProjectIteration getTargetVersion() {
+        if(targetVersion == null) {
+            targetVersion =
+                    projectIterationDAO.getBySlug(targetProjectSlug,
+                            targetVersionSlug);
+        }
+        return targetVersion;
+    }
+
+    public HProject getSourceProject() {
+        if(sourceProject == null) {
+            sourceProject = projectDAO.getBySlug(sourceProjectSlug);
+        }
+        return sourceProject;
+    }
+
+    public List<HProjectIteration> getSourceVersions() {
+        List<HProjectIteration> versions =
+                getSourceProject().getProjectIterations();
+
+        if(versions.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        //remove target version if both are the same project or obsolete
+        if (StringUtils.equals(sourceProjectSlug, targetProjectSlug)) {
+            for (HProjectIteration version : versions) {
+                if (version.getSlug().equals(targetVersionSlug)
+                        || version.getStatus().equals(EntityStatus.OBSOLETE)) {
+                    versions.remove(version);
+                    break;
+                }
+            }
+        }
+        return versions;
+    }
+
+    public List<HProject> getProjects() {
+        return projectDAO.getProjects(EntityStatus.ACTIVE,
+                EntityStatus.READONLY);
+    }
+
+    public void startMergeTranslations() {
+        if (StringUtils.isEmpty(sourceProjectSlug)
+                || StringUtils.isEmpty(sourceVersionSlug)
+                || StringUtils.isEmpty(targetProjectSlug)
+                || StringUtils.isEmpty(targetVersionSlug)) {
+            FacesMessages.instance().add(StatusMessage.Severity.WARN,
+                msgs.get("jsf.iteration.mergeTrans.noSourceAndTarget"));
+            return;
+        }
+        if (isCopyActionsRunning()) {
+            FacesMessages.instance().add(StatusMessage.Severity.WARN,
+                msgs.get("jsf.iteration.mergeTrans.hasCopyActionRunning"));
+            return;
+        }
+        mergeTranslationsManager.startMergeTranslations(sourceProjectSlug,
+                sourceVersionSlug, targetProjectSlug, targetVersionSlug,
+                !keepExistingTranslation);
+    }
+
+    // Check if copy-trans, copy version or merge-trans is running for given
+    // version
```

Comment:
"for given version" suggests that you can give a version as the argument. I suggest "for the target version" instead.